features/step_definitions/data_set_steps.rb

```diff
@@ -115,8 +137,16 @@
   save_and_open_page
 end
 
-Then /^I fill in the form to create the "(.*?)" service$/ do |name|
-  fill_in_form_with(name, csv_path_for_data(name))
+When /^I fill out the form with the following attributes to create a service:$/ do |table|
+  params = table.rows_hash.symbolize_keys!
+  assert params[:name]
```

Comment:
Your intention is fine, however, it would be better to raise an `ArgumentError` instead.  Not passing in a `:name` param is a failure of the test-harness code, rather than a failure of the code under test.  An `assert` will fail with a test failure, whereas an `ArgumentError` will cause a test error and while the result is the same (the test doesn't pass) the semantic meaning of why is different and should point the developer at the correct place to fix.